orchestrator/websocket/managers/memory_websocket_manager.py

```diff
@@ -68,7 +68,7 @@ async def broadcast_data(self, channels: List[str], data: Dict) -> None:
     async def remove_ws(self, websocket: WebSocket, channel: str) -> None:
         if websocket.client_state != WebSocketState.DISCONNECTED:
             await self.disconnect(websocket)
-        if channel in self.connections_by_pid:
+        if channel in self.connections_by_pid and websocket in self.connections_by_pid[channel]:
             self.connections_by_pid[channel].remove(websocket)
             if len(self.connections_by_pid[channel]):
```

Comment:
I guess this code is just used for testing, right? I find it difficult to follow the logic. First we remove the websocket from the list of this channel, and then we delete the channel if it still contains elements? Why do we have to remove it from the channel in the first place? Timing issues? Other?